deploy.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# -------- Build Settings -------- #
+# directory to deploy
+BUILD_DIR="./dist"
+# directory on target server
+DEPLOY_DIR="./latest"
+# use to install dependencies without disruption
+TEMP_DIR="./staged"
+# server to connect to
+TARGET="hanbot@hanbot.elviswolcott.com"
+# file to run
+ENTRY="index.js"
+# name to give job (mostly for debugging on server)
+JOB_NAME="hanbot"
+
+# --------- Create .env ---------- #
+echo "\
+DISCORD_WELCOME_CHANNEL=${DISCORD_WELCOME_CHANNEL}
+DISCORD_ADMITTED_ROLE=${DISCORD_ADMITTED_ROLE}
+DISCORD_ACTIVE_ROLE=${DISCORD_ACTIVE_ROLE}
+DISCORD_TOKEN=${DISCORD_TOKEN}
+ZOOM_API_KEY=${ZOOM_API_KEY}
+ZOOM_API_SECRET=${ZOOM_API_SECRET}
+ZOOM_WEBHOOK_SCRET=${ZOOM_WEBHOOK_SECRET}
+ZOOM_MEETING_ID=${ZOOM_MEETING_ID}
+ZOOM_TIME_THRESHOLD=${ZOOM_TIME_THRESHOLD}
+ZOOM_TIME_ANNOUNCEMENT_CHANNEL=${ZOOM_TIME_ANNOUNCEMENT_CHANNEL}
+TRAVIS_BUILD_NUMBER=${TRAVIS_BUILD_NUMBER}
+TRAVIS_COMMIT=${TRAVIS_COMMIT}
+" >> ".env"
+
+
+# --------- Deploy Steps --------- #
+# remove extra files
+find "${BUILD_DIR}" -name "*.map" -type f -delete
+rm "${BUILD_DIR}/.tsBuildInfo"
+# transfer the build to the target
+rsync -r --delete-after --quiet "${TRAVIS_BUILD_DIR}/${BUILD_DIR}" "${TARGET}:${TEMP_DIR}"
+# bring over package.json & .env
+rsync -r --delete-after --quiet "${TRAVIS_BUILD_DIR}/package.json" "${TARGET}:${TEMP_DIR}/package.json"
+rsync -r --delete-after --quiet "${TRAVIS_BUILD_DIR}/.env" "${TARGET}:${TEMP_DIR}/.env"
```

Comment:
Why are you deploying it piecemeal this way? This is a pretty small project, you should be fine to `rsync` the entire directory (maybe ignore `node_modules` for architecture issues). This seems to be adding needless complexity and additional points of failure.

If you must deploy only the relevant files, you shouldn't change the relative directory structure (ie. you should end up with `/package.json` and `/dist/index.js`).

Reply:
```
hanbot@ctl:~/latest$ ls
dist  node_modules  package-lock.json  package.json
```
This doesn't change the directory structure. Somewhat unintuitively , `rsync /path/to/dir` creates `dir` in the destination unless you add the trailing `/`.

As far as the syncing process it's not super relevant for a project of this size, but the goal is to ensure that only build artifacts are sent to prod.